tests/route-spec/responses.test.ts

```diff
@@ -0,0 +1,278 @@
+import test from "ava"
+import { createWithEdgeSpec } from "src/create-with-edge-spec"
+import { EdgeSpecRequest, EdgeSpecResponse } from "src/types/web-handler"
+import { z } from "zod"
+
+test("serializes json basic", async (t) => {
+  const withEdgeSpec = createWithEdgeSpec({
+    apiName: "hello-world",
+    productionServerUrl: "https://example.com",
+
+    authMiddlewareMap: {},
+    globalMiddlewares: [],
+  })
+
+  const response = await withEdgeSpec({
+    auth: "none",
+    methods: ["GET"],
+    jsonResponse: z.object({
+      hello: z.string(),
+    }),
+  })(() => {
+    return EdgeSpecResponse.json({
+      hello: "world",
+    })
+  })({} as EdgeSpecRequest)
+
+  t.is(response.status, 200)
+  t.is(response.headers.get("content-type"), "application/json")
+  t.is(await streamToString(response.body), JSON.stringify({ hello: "world" }))
```

Comment:
I think

```suggestion
  t.is(await response.text(), JSON.stringify({ hello: "world" }))
```

would work too

Reply:
oh